### ISSUE TITLE
Compute norm consistently everywhere

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#1433]
Interestingly, the last approach is faster!

```
In [7]: np.sqrt(u @ u)
Out[7]: 0.9193857259765762

In [8]: norm(u)
Out[8]: 0.9193857259765762

In [9]: (u @ u) ** .5
Out[9]: 0.9193857259765762

In [10]: %timeit np.sqrt(u @ u)
1.32 µs ± 5.52 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [11]: %timeit norm(u)
2.16 µs ± 21.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [12]: %timeit (u @ u) ** .5
708 ns ± 7.26 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [13]: def unorm(u):
    ...:     return (u @ u) ** .5
    ...: 

In [14]: %timeit unorm(u)
753 ns ± 16.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

With numba though, the differences are not so big:

```
In [15]: from numba import njit

In [16]: @njit
    ...: def unorm1(u):
    ...:     return np.sqrt(u @ u)
    ...: 

In [17]: @njit
    ...: def unorm2(u):
    ...:     return norm(u)
    ...: 
    ...: 

In [18]: @njit
    ...: def unorm3(u):
    ...:     return (u @ u) ** .5
    ...: 

In [19]: unorm1(u)
Out[19]: 0.9193857259765762

In [20]: unorm2(u)
Out[20]: 0.9193857259765762

In [21]: unorm3(u)
Out[21]: 0.9193857259765762

In [22]: %timeit unorm1(u)
172 ns ± 1.05 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [23]: %timeit unorm2(u)
299 ns ± 9.15 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [24]: %timeit unorm3(u)
203 ns ± 2.16 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

But I agree we should be consistent:

1. Add a fast `norm` implementation in `poliastro.core.util` (or even the new upcoming `poliastro._math`)
2. Use it everywhere

_Originally posted by @astrojuanlu in https://github.com/poliastro/poliastro/pull/1430#discussion_r775476935_

**Comment by Yash-10** [comment]
This is a great observation, and a neat change to include! Now that `poliastro._math` has been created, maybe we could include it there itself as you suggested.

While thinking of this a bit, I wondered what could be done to:

https://github.com/poliastro/poliastro/blob/9797eb079d7ce86b39aaef379b750d88c3135ce2/src/poliastro/util.py#L13-L22

which is in the high level `poliastro.util` module.

Should this high-level `norm` use the fast norm implementation or could we remove the high level `norm` completely and instead do an equivalent transformation?

**Comment by astrojuanlu** [comment]
Good point @Yash-10 ! What I'd do is replace `norm_np` with our own `poliastro._math.linalg.norm_fast`. Which, according to my benchmarks above, should be `unorm1`.

**Comment by astrojuanlu** [comment]
Fixed in #1442!
